### DISCUSSION TITLE
[Chapter 10] Cashable의 대상 선정과 수정의 번거로움

### DISCUSSION THREAD

**Discussion by ahngilwoong** [cowaine/spring-boot-for-msa#167] [Chapter 10. Redis and Spring Cache] [answered]
1. Cashable을 통해 효율적인 조회가 가능한 것으로 파악했습니다.
하지만 무분별한 Cashable 사용은 안될 것 같은데 주의 조건들이 뭐가 있을지 얘기를 나눠보고 싶습니다.

2. CashPut과 CashEvit를 통해 캐시의 수정 삭제가 가능한 것으로 확인했습니다.
그러면 UD의 작업을 할 떄마다 번거롭게 저렇게 써줘야하는지 DB가 업데이트 되었을때 자동으로 캐시도 업데이트 되도록 하는 설정이 따로
있는지 의논해보고 싶습니다. 또 각자의 cashable의 적절한 사용 예시가 있다면 사연을 공유받고싶습니다.

**Comment by corock** [comment, marked as answer]
1. 데이터의 추가/수정/삭제 작업이 많은 대상에 대해서는 캐싱 사용을 지양해야 하지 않나 싶습니다.
2. 저 같은 경우 경험한 사례로 쇼핑몰에서 장바구니에 담은 상품 카운트를 레디스 캐시를 이용해서 사용자에게 보여주고 있습니다.
